aten/src/ATen/native/Assert.cpp

```diff
@@ -0,0 +1,25 @@
+#include <cstddef>
+#include <string>
+
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+namespace native {
+
+void _assert(bool condition, std::string message) {
+  if (!condition) {
+    throw c10::AssertionError(message.c_str(), "");
+  }
+}
+
+void _assert(const Tensor& condition, std::string message) {
```

Comment:
what is the behavior when `condition` is not a Scalar? or do we require the input to be a Scalar? can we add assert/doc for this?

Reply:
the `condition.is_nonzero()` function used later on in this function verifies that the tensor has a single element and throws an error otherwise.  So, for now we are piggybacking on those checks.  Lmk if we need something stronger.

Reply:
sounds good, please add a comment thanks!